chalice/awsclient.py

```diff
@@ -483,3 +483,29 @@ def _client(self, service_name):
             self._client_cache[service_name] = self._session.create_client(
                 service_name)
         return self._client_cache[service_name]
+
+    def add_permission_for_authorizer(self, rest_api_id, function_arn,
+                                      random_id):
```

Comment:
On second thought, I think it would make sense to parameterize it but not completely hide it.  This is used to ensure uniqueness and makes it impossible to do retries at any level above the awsclient level.  I'd propose making this an `Optional[str]`.